### PR TITLE
FIX #318

### LINKED ISSUE #318: `ExecuteTARunOld` does not always return a `StatusType`
If the return string of the TA includes a "wrong" status (e.g., because of a typo),
`ExecuteTARunOld` does not check this and returns a string as status type and not an instance of `StatusType`. Afterwards SMAC performs a pure random search without complaining.

TODO: the parsing+checking in `ExecuteTARunOld` has to be stricter.

### PR DESCRIPTION
**Pull request by mlindauer**
Proper parsing of status returned by TA.

+ some restructure for better unit tests

### CODE REVIEW

**Comment by mlindauer** [comment]
@AndreBiedenkapp could you please review this PR.

File: smac/tae/execute_ta_run_old.py [hunk 1]
```diff
@@ -94,16 +84,18 @@ def run(self, config, instance=None,
...
+        elif status.upper() in ["CRASHED"]:
             status = StatusType.CRASHED
-        elif status in ["ABORT"]:
+        elif status.upper() in ["ABORT"]:
             status = StatusType.ABORT
-        elif status in ["MEMOUT"]:
+        elif status.upper() in ["MEMOUT"]:
             status = StatusType.MEMOUT
+        else:
+            status = StatusType.CRASHED
```
Comment (AndreBiedenkapp) [inline smac/tae/execute_ta_run_old.py]: I'm not sure if we want the status to be crashed if the output of the target algorithm is unintelligible.

File: smac/tae/execute_ta_run_old.py [hunk 2]
```diff
@@ -94,16 +84,18 @@ def run(self, config, instance=None,
...
 
         if status in [StatusType.CRASHED, StatusType.ABORT]:
             self.logger.warn(
```
Comment (AndreBiedenkapp) [inline smac/tae/execute_ta_run_old.py]: Especially if we display the  warning like that.
As a user of SMAC I'd then expect that my algorithm crashed and would not start checking for typos or something like that.

SMAC should see the status as crashed but the user needs some more information about what went wrong. Maybe we could add a logging warning for unintelligible output without introducing a new StatusType?

File: smac/tae/execute_ta_run_old.py (hunk 1 above)
Comment (mlindauer) [reply smac/tae/execute_ta_run_old.py]: Thanks for pointing this out; added warn logging message.

**Pull request merged**